### DISCUSSION TITLE
Installing FastAPI without Cryptography (Rust issue)

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#6457] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Not applicable.
```

### Description

We are trying to install FastAPI on an IoT device. We ran into the problem with the `cryptography` package that is required by FastAPI.  We can't install Rust on the device. From what I found out, the `cryptography` package is used mainly for `HTTPS`, which we won't be using. Is it possible somehow to add a flag to FastAPI installation through pip, to skip the `cryptography` package?

### Operating System

Linux

### Operating System Details

Yocto package for Linux distributions.

### FastAPI Version

0.70.0

### Python Version

3.9

**Comment by yezz123** [comment]
Most of the HTTP & Authentication or security functions are based on this, that's why I think you can fork the project and try your changes on it by deleting all functionality and unused packages and building the package locally.

**Comment by K3UL** [comment]
It is not strictly what I would call IoT, so I will add my use case too : I have this problem with the package on a Raspberry PI with the armv7 architecture, which is pretty widespread.

**Comment by Garito** [comment]
IMO always good to have clear separation between components

**Comment by K3UL** [comment]
Would it be a bad practice/complicated/impossible or anything else, to have a prebuilt `cryptography` ? I honestly find ridiculous that I need the whole Rust toolchain just to install the dependency of a dependency.

**Comment by thealtofwar** [comment, marked as answer]
> Would it be a bad practice/complicated/impossible or anything else, to have a prebuilt `cryptography` ? I honestly find ridiculous that I need the whole Rust toolchain just to install the dependency of a dependency.

You can look for prebuilt wheels on 
https://pypi.org/simple/cryptography/
If there's nothing, you have to build it yourself or on a similar machine which does have Rust.

**Comment by ghost** [comment]
Following @thealtofwar suggestion, we created a similar setup using Docker and created the wheel there. Thanks!
